compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -301,6 +301,9 @@ pub fn eval_in_interpreter<'mir, 'tcx>(
     cid: GlobalId<'tcx>,
     is_static: bool,
 ) -> ::rustc_middle::mir::interpret::EvalToAllocationRawResult<'tcx> {
+    // `is_static` just means "in static", it could still be a promoted!
+    debug_assert_eq!(is_static, ecx.tcx.static_mutability(cid.instance.def_id()).is_some());
```

Comment:
@oli-obk the `is_static` parameter here seems redundant, I couldn't quite figure out why we have it?

Reply:
😆 just to avoid a second `is_static` invocation.